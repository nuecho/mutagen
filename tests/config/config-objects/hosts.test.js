const checkKey = (cfgObject) => {return cfgObject.name === "cfgObjectTest" && cfgObject.tenant === "Environment"};

cfgObjectTests("hosts", "hosts-config.json");
