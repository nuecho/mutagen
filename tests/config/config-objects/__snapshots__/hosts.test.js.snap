// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[mutagen hosts config import-export] should create the hosts: the hosts have been created 1`] = `
Object {
  "stderr": "",
  "stdout": "+ host [cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;

exports[`[mutagen hosts config import-export] should skip the same hosts: the hosts have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "= host [cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;

exports[`[mutagen hosts config import-export] should throw exception when new hosts to import miss mandatory properties hosts: exception has been thrown 1`] = `
Object {
  "stderr": "Validation failed.
Missing properties:
  - host [cfgObjectTest]:
    - lcaPort
    - osInfo
    - type
",
  "stdout": "",
}
`;

exports[`[mutagen hosts config import-export] should throw exception when updating hosts with different values for unchangeable properties: exception has been thrown 1`] = `
Object {
  "stderr": "Validation failed.
Unchangeable properties:
  - host [cfgObjectTest]:
    - type
",
  "stdout": "",
}
`;

exports[`[mutagen hosts config import-export] should update the modified hosts: the modified hosts have been updated 1`] = `
Object {
  "stderr": "",
  "stdout": "~ host [cfgObjectTest]
Completed. 1 object(s) imported.
",
}
`;
